### PR TITLE
chore(deps):  use sigs.k8s.io/yaml

### LINKED ISSUE #5192: Choose a library for encoding YAML
### What happened?

All over the project we have 60 matches for `"github.com/ghodss/yaml"` and 5 matches for `"gopkg.in/yaml.v2"`. Ideally, we'd like to have a single library for this.

### PR DESCRIPTION
**Pull request by mmorel-35**
Closes #5192
Signed-off-by: Matthieu MOREL <matthieu.morel35@gmail.com>

### Checklist prior to review

- [x] Link to docs PR or issue --
- [x] Link to UI issue or PR --
- [x] Is the [issue worked on linked][1]? --
- [x] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [x] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Unit Tests --
- [x] E2E Tests --
- [x] Manual Universal Tests --
- [x] Manual Kubernetes Tests --
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [x] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/config/util.go [hunk 1]
```diff
@@ -22,5 +21,5 @@ func ToJson(cfg Config) ([]byte, error) {
 		return nil, err
 	}
 	// there is no easy way to convert yaml to json using gopkg.in/yaml.v2
-	return ghodss_yaml.YAMLToJSON(yamlBytes)
+	return yaml.YAMLToJSON(yamlBytes)
 }
```
Comment (lahabana) [inline pkg/config/util.go:25]: Sounds like if we were to switch from `yaml` tags to `json` tags everywhere in config we'd be able to get rid of this entire package and more importantly this double serialization right?

If that's the case could you please open an issue to do this. Thanks!

**Review by lahabana — APPROVED** [review]
Thanks a lot for this contrib! It's a nice cleanup!

We're just about to cut the 2.0 branch and while I'm pretty sure this change is not going to cause issues I'd rather not take the risk that late in the release.
So I'll wait until we cut the 2.0 branch to merge this PR into master if that's ok?

**Comment by mmorel-35** [comment]
There is an issue : https://github.com/kubernetes-sigs/yaml/issues/64 that was not treat so I don't think that sigs.k8s.io/yaml might be able able to replace gopkg.in/yaml.v2

**Comment by lahabana** [comment]
I guess we could apply what's mentioned: https://stackoverflow.com/a/48051946 there's even already an apimachinery type for this: https://github.com/kubernetes/apimachinery/blob/df993592a122931b8aac4db57689e09458a2332c/pkg/apis/meta/v1/duration.go

File: pkg/config/access/config.go [hunk 2]
```diff
@@ -47,9 +47,9 @@ func DefaultAccessConfig() AccessConfig {
 // AccessConfig defines a configuration for acccess control
 type AccessConfig struct {
 	// Type of the access strategy (available values: "static")
-	Type string `yaml:"type" envconfig:"KUMA_ACCESS_TYPE"`
+	Type string `yaml:"type" json:"type" envconfig:"KUMA_ACCESS_TYPE"`
```
Comment (lahabana) [inline pkg/config/access/config.go:50]: Shouldn't we be dropping the `yaml` key as well?

File: pkg/config/util.go (hunk 1 above)
Comment (mmorel-35) [reply pkg/config/util.go:25]: It's done :) !

File: pkg/config/app/kuma-dp/testdata/default-config.golden.yaml [hunk 3]
```diff
@@ -1,19 +1,20 @@
...
-  caCert: ""
-  caCertFile: ""
+  url: https://localhost:5678
 dataplane:
-  proxyType: dataplane
   drainTime: 30s
+  proxyType: dataplane
 dataplaneRuntime:
   binaryPath: envoy
+  resources: {}
```
Comment (lahabana) [inline pkg/config/app/kuma-dp/testdata/default-config.golden.yaml:13]: This is strange especially when the `omitempty` tag is present

**Review by lahabana — APPROVED** [review]

**Comment by lahabana** [comment]
updated to latest and setup auto-merge once CI passes! Thanks for the work @mmorel-35

File: pkg/config/app/kuma-dp/testdata/default-config.golden.yaml (hunk 3 above)
Comment (mmorel-35) [reply pkg/config/app/kuma-dp/testdata/default-config.golden.yaml:13]: I don't know why, it happens, this is the result of golden files generation

**Pull request merged**